Common/OvermorrowGlobalNPC.cs

```diff
@@ -208,7 +199,7 @@ public override void OnHitByProjectile(NPC npc, Projectile projectile, int damag
                     {
                         for (int i = 0; i < 2; i++)
                         {
-                            Projectile.NewProjectile(npc.Center, Vector2.One.RotatedByRandom(MathHelper.TwoPi) * 4, ModContent.ProjectileType<IchorStream>(), 12, 2f, owner.whoAmI);
+                            Projectile.NewProjectile(projectile.GetProjectileSource_FromThis(), npc.Center, Vector2.One.RotatedByRandom(MathHelper.TwoPi) * 4, ModContent.ProjectileType<IchorStream>(), 12, 2f, owner.whoAmI);
```

Comment:
Hundreds of these changes as well. Spawning projectiles and NPCs now use an "Entity source". It _can_ be null, but is usually easy to set. This is described in the migration guide.